src/knownIssues/Test-SdnKIHostRootStoreNonRootCert.ps1

```diff
@@ -0,0 +1,104 @@
+        Type the NetBIOS name, an IP address, or a fully qualified domain name of one or more remote computers.
+    .PARAMETER Credential
+        Specifies a user account that has permission to perform this action. The default is the current user.
+    .PARAMETER NcRestCredential
+        Specifies a user account that has permission to access the northbound NC API interface. The default is the current user.
+    .EXAMPLE
+        PS> <sample 1>
+    .EXAMPLE
+        PS> <sample 2>
+    #>
+
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        [System.String[]]$ComputerName = $global:SdnDiagnostics.EnvironmentInfo.Server,
+
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+    )
+
+    try {
+        "Validating Certificates under Root CA Store" | Trace-Output
+
+        if($null -eq $ComputerName){
+            throw New-Object System.NullReferenceException("Please specify ComputerName parameter or execute Get-SdnInfrastructureInfo to populate environment details")
+        }
+
+        # if Credential parameter not defined, check to see if global cache is populated
+        if(!$PSBoundParameters.ContainsKey('Credential')){
+            if($Global:SdnDiagnostics.Credential){
+                $Credential = $Global:SdnDiagnostics.Credential
+            }    
+        }
+
+        # if NcRestCredential parameter not defined, check to see if global cache is populated
+        if(!$PSBoundParameters.ContainsKey('NcRestCredential')){
+            if($Global:SdnDiagnostics.NcRestCredential){
+                $NcRestCredential = $Global:SdnDiagnostics.NcRestCredential
+            }    
+        }
+
+        $issueDetected = $false
+        $arrayList = [System.Collections.ArrayList]::new()
+        
+        $scriptBlock = {
+            $nonRootCerts = [System.Collections.ArrayList]::new()
+            $rootCerts = Get-ChildItem Cert:LocalMachine\Root
+            foreach ($rootCert in $rootCerts) {
+                if ($rootCert.Subject -ne $rootCert.Issuer) {
+                    $certInfo = [PSCustomObject]@{
+                        Thumbprint = $rootCert.Thumbprint
+                        Subject    = $rootCert.Subject
+                        Issuer     = $rootCert.Issuer
+                    }
+                    [void]$nonRootCerts.Add($certInfo)
+                }
+            }
+            return $nonRootCerts
+        }
+
+        foreach($node in $ComputerName){
+            $nonRootCerts = Invoke-PSRemoteCommand -ComputerName $ComputerName -Credential $Credential -ScriptBlock $scriptBlock -PassThru
```

Comment:
You are within a foreach block but submitting the command to `$ComputerName` which contains all the nodes. Should this be `$node`? Or should this command be moved outside the foreach block?

Reply:
Yup.. should be $node. Fixed